### DISCUSSION TITLE
Control over history method

### DISCUSSION THREAD

**Discussion by ethmcc** [swup/swup#357] [Feature ideas]
First of all, thanks for the great work on this product! I have an idea for a feature I wonder if you'd consider:

- It would be helpful to me if there were some attribute I could set on a link to tell Swup not to push a history item when I follow that link.

When developing websites, I often have routes which are intended mainly to update content on the current page and redirect back to the current page.

Consider something like this pseudocode Flask route:
```python
@app.route("/update/<item_id>", methods=["GET", "POST"])
def update_item(item_id):
    update_stuff_in_database(item_id)
    redirect(url_for("app.index")
```

If I'm visiting my site's `/` route and click on a link for `/update/<item_id>`, Swup will helpfully push a history entry, because it sees that I'm navigating to a different page. But it would be nice if I could override that behavior in cases like this, where I know the route will redirect to the current page.

-------

<details><summary>I have made a fork and whipped together something basic for my needs (expand for details):  </summary>

In this implementation, if `loadPage` is called with `noPushHistory` set in `data`, then the `createHistoryRecord` subroutine is skipped. To use this option in normal use outside of direct API calls to `loadPage`, the user sets `optionPushHistory : true` in options, then when a link is clicked the fork checks if the link has an attribute `data-swup-no-push-history` and calls `loadPage` with or without the `noPushHistory` option as appropriate. 

If `loadPage` is called as normal without `noPushHistory`, or if Swup is initialized without the `optionPushHistory` option, then they behave as usual.
</details>

[https://github.com/ethmcc/swup/commit/855945...](https://github.com/ethmcc/swup/commit/855945643079b537255e07f08275d17e18782bae)

My apologies if there is already some easy way to accomplish this sort of thing using the API as-is, I couldn't see any way to do it. And similarly I couldn't see any way to implement this feature through the existing plugins framework.

**Comment by daun** [comment]
Since swup 3.1, you can tell swup to replace the history entry instead of creating a new one:

```js
<a href="/about" data-swup-history="replace">About</a>
```
